sse/src/main/scala/akka/stream/alpakka/sse/scaladsl/EventSource.scala

```diff
@@ -94,11 +94,9 @@ object EventSource {
         }
         send(request).flatMap(Unmarshal(_).to[EventSource]).fallbackTo(noEvents)
       }
-      def recover(eventSource: EventSource) = eventSource.recoverWithRetries(1, { case _ => Source.empty })
-      def delimit(eventSource: EventSource) = eventSource.concat(singleDelimiter)
-      Flow[Option[String]]
-        .mapAsync(1)(getEventSource)
-        .flatMapConcat((recover _).andThen(delimit))
+      def recoverAndDelimit(eventSource: EventSource) =
+        eventSource.recoverWithRetries(0, { case _ => Source.empty }).concat(singleDelimiter)
+      Flow[Option[String]].mapAsync(1)(getEventSource).flatMapConcat(recoverAndDelimit)
```

Comment:
Ah, my bad, didn't know that behaviour with retries `0`, which is somewhat useless, as it never retries but just fails the stage with the exception upstream failed with it makes more sense to just remove the recovery. `Source.failed(ex)` will give you the same behaviour as `Source.failed(ex).recoverWithRetries(0, { ... whatever ...})`.